### PR TITLE
bpo-34488: Improve performance of BytesIO.writelines() by avoiding creation of unused ints.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://bugs.python.org/issue34488

### CODE REVIEW

File: Modules/_io/bytesio.c [hunk 1]
```diff
@@ -155,16 +172,31 @@ resize_buffer(bytesio *self, size_t size)
...
-    assert(len >= 0);
+    if (check_closed(self)) {
+        return -1;
+    }
+    if (check_exports(self)) {
+        return -1;
+    }
+
+    Py_buffer buf;
+    if (PyObject_GetBuffer(b, &buf, PyBUF_CONTIG_RO) < 0) {
```
Comment (ZackerySpytz) [inline Modules/_io/bytesio.c:189]: It seems `buf` is leaked in some cases below.

Comment (sir-sigurd) [reply Modules/_io/bytesio.c:189]: Thanks for remark, I think I fixed it now.

File: Modules/_io/bytesio.c [hunk 2]
```diff
@@ -30,17 +30,34 @@ typedef struct {
   * exports > 0.  Py_REFCNT(buf) == 1, any modifications are forbidden.
 */
 
+static int
```
Comment (pablogsal) [inline Modules/_io/bytesio.c:33]: Should these be marked as inlined?

Comment (methane) [reply Modules/_io/bytesio.c:33]: Compilers can inline the function even if it is not marked as `inline`.
No need to mark `inline` explicitly unless the very critical part.

**Comment by methane** [comment]
LGTM.  Would you add NEWS file?

**Pull request merged**